### PR TITLE
Initial thrift configuration

### PR DESCRIPTION
**Pull request by pnovotnak**
This allows configuring thrift features via mesh config. It it is a response to comments on https://github.com/istio/istio/pull/19852

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -387,8 +387,16 @@ message MeshConfig {
   // Configure the provision of certificates.
   repeated Certificate certificates = 47;
 
+  // Specify thrift rate limit service URL. If pilot has thrift protocol support enabled,
+  // this will enable the rate limit service for destinations that have matching rate
+  // limit configurations.
+  string thrift_rate_limit_url = 49;
+
+  // Specify thrift rate limit service timeout, in milliseconds
+  string thrift_rate_limit_timeout = 50;
```
Comment (rshriram) [inline mesh/v1alpha1/config.proto]: convert this to a message please - something specific for Thrift where you can pile in all these configs.

File: mesh/v1alpha1/config.proto [hunk 2]
```diff
@@ -387,6 +387,16 @@ message MeshConfig {
   // Configure the provision of certificates.
   repeated Certificate certificates = 47;
 
+  message ThriftConfig {
```
Comment (howardjohn) [inline mesh/v1alpha1/config.proto]: cc @istio/wg-policies-and-telemetry-maintainers

File: mesh/v1alpha1/config.proto [hunk 3]
```diff
@@ -387,6 +387,16 @@ message MeshConfig {
...
+    // Specify thrift rate limit service URL. If pilot has thrift protocol support enabled,
+    // this will enable the rate limit service for destinations that have matching rate
+    // limit configurations.
+    string thrift_rate_limit_url = 1;
```
Comment (rshriram) [inline mesh/v1alpha1/config.proto]: nit: you dont need thrift_ since this is under the ThriftConfig.

File: mesh/v1alpha1/config.proto [hunk 4]
```diff
@@ -387,6 +387,16 @@ message MeshConfig {
...
+
+    // Specify thrift rate limit service timeout, in milliseconds
+    string thrift_rate_limit_timeout = 2;
```
Comment (rshriram) [inline mesh/v1alpha1/config.proto]: Do you want this to be in string? shouldn't it be in protobuf.Duration or whatever? And is there a default?

**Pull request merged**